compiler/rustc_hir_typeck/src/_match.rs

```diff
@@ -139,7 +139,30 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 &cause,
                 Some(&arm.body),
                 arm_ty,
-                |err| self.suggest_removing_semicolon_for_coerce(err, expr, arm_ty, prior_arm),
+                |err| {
+                    if let hir::ExprKind::Block(block, _) = arm.body.kind
+                        && let Some(expr) = block.expr
+                        && let arm_tail_ty = self.node_ty(expr.hir_id)
+                        && arm_tail_ty.is_never()
+                        && !arm_ty.is_never()
+                    {
+                        err.span_label(
+                            expr.span,
+                            format!(
+                                "this expression is of type `!`, but it get's coerced to \
```

Comment:
"get's" -> "gets", but i think "is" is equivalent here

```suggestion
                                "this expression is of type `!`, but it is coerced to \
```